File: src/main/java/io/github/takenoha/towerdefense/paper/RaidSealAutomationPolicy.java
package io.github.takenoha.towerdefense.paper;

/** Pure policy helpers for paths that can otherwise bypass player-facing seal safeguards. */
public final class RaidSealAutomationPolicy {
    private RaidSealAutomationPolicy() {
    }

    public static boolean cancelRightClick(
            boolean validSeal, boolean rightClick) {
        return validSeal && rightClick;
    }

    public static boolean cancelCrafter(
            boolean pluginRecipe,
            boolean resultIsTemplate,
            boolean echoShardIngredient,
            boolean legacyEnderEyeIngredient) {
        return pluginRecipe || resultIsTemplate
                || echoShardIngredient
                || legacyEnderEyeIngredient;
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/RaidSealAutomationPolicy.kt
package io.github.takenoha.towerdefense.paper

/** Pure policy helpers for paths that can otherwise bypass player-facing seal safeguards. */
class RaidSealAutomationPolicy private constructor() {
    companion object {
        @JvmStatic
        fun cancelRightClick(validSeal: Boolean, rightClick: Boolean): Boolean =
            validSeal && rightClick

        @JvmStatic
        fun cancelCrafter(
            pluginRecipe: Boolean,
            resultIsTemplate: Boolean,
            currentSealIngredient: Boolean,
            legacySealIngredient: Boolean,
        ): Boolean =
            pluginRecipe || resultIsTemplate || currentSealIngredient || legacySealIngredient
    }
}
